### ISSUE TITLE
Improvements for new static site

### ISSUE THREAD

**Issue by cmungall** [geneontology/go-site#789]
Required:

 * [x] decide/implement basic theme plus frontmatter and templates. TODO: decide page vs doc @lpalbou @kltm 
   * it would be advantageous for backwards compat if the 'docs' path was instead 'page' (as in the old site)--some links not need to be mapped and we'd have less confusion with things like https://github.com/geneontology/go-site/issues/811
 * [x] decide on contents we'd like to keep from old website @selewis 
 * [x] set up http / forwarding @kltm @lpalbou (mixture of apache and gh pages)
 * [x] redirect http://geneontology.org/ontology/ to PURLs @cmungall 
 * [x] redirect for now for /go-cam @kltm 
 * [x] document SOP for GOC members to update pages @lpalbou 
 * [x] get feedback from product owner @selewis 
 * [x] get feedback from GOC members @selewis 
 * [x] merge search boxes  @lpalbou @kltm 
 * [x] deploy gh pages live @kltm 
 * [x] <strike>custom 404 page (possibly search) @lpalbou @kltm </strike> partially on #978
 * [x] <strike>check all core URLs still work @lpalbou @kltm </strike> now #978
 * [x] <strike>check URLs via spidering  @lpalbou @kltm  </strike> now #978
 * [x] <strike>some kind of site document search (google?)</strike> now #977
 * [ ] <strike>kill drupal @kltm </strike> not necessary for release
 * [x] <strike>flip DNS and redeploy @kltm </strike> now #958 
 * [x] <strike>check logging and backups @kltm </strike> now #958 

Bugs:
 * [x] <strike>Homepage mini-header w/current release needs to display current release; is on Oct 2018: https://test.geneontology.io/ </strike> now #979
* [x] Browse Tools, Browse Guides links on homepage are inactive
* [x] Broken link (https://test.geneontology.io/docs/publications/) on https://test.geneontology.io/docs/introduction-to-go-resource/
* [x] Security warning when submitting EA from homepage
* [x] Broken link: https://test.geneontology.io/docs/collaborations/cjmungall@lbl.gov on https://test.geneontology.io/docs/collaborations/
* [x] https://test.geneontology.io/docs/collaborations/ lists SF tracker as Cell Type Ontology (CL) contact, this seems slightly OOD- should be https://github.com/obophenotype/cell-ontology/issues/new perhaps??
* [x] Drop down titles should match page titles.  The don't, currently (i.e. contributing-to-go-terms

Possible minor issues:
* [x] Consistency of sentence case: All pages need to be checked for compliance, include page titles? 
* [x] https://test.geneontology.io/docs/introduction-to-go-resource has link to https://geneontology.github.io/docs/go-enrichment-analysis/ , which seems to be at a different level of text zoom? I'd like to look at this again on a 'final' version of the site.
* [x] https://test.geneontology.io/docs/collaborations/ has link to http://geneontology.org/page/syngo-synapse-biology, which has old skin
* [x] I'd like to simplify the "Past collaborations" section on https://test.geneontology.io/docs/collaborations/ : remove the "Publication available here" phrase and just make e.g. Autophagy a link. Anything wrong with that, other than tiny links are hard to access on some mobile devices (accessibility concerns)? Any input?
* [x] Many external links to lists of publications. Some go to pubmed (references to 140,000 published papers), some go to EuropePMC (all on https://test.geneontology.io/docs/literature/), any logic/reasoning? Should this be consistent?
* [ ] Still hard links to http://geneontology.org/external2go, eg http://geneontology.org/external2go/um-bbd_enzymeid2go : see https://test.geneontology.io/docs/download-mappings/: waiting on https://github.com/geneontology/go-site/issues/688
* [x] Help page (http://help.geneontology.org/) is not in .md; should it be? Also, has hard links to "old" site: mailing list, Contributing to GO, faq
* [x] link ECs to ECO, suggested in https://github.com/geneontology/geneontology.github.io/pull/114 esp on guide-go-evidence-codes.md

Bonus:

 - integration of various md/yamldown products from go-site
    - gorefs
    - gorules
    - xrefs yaml
- add new content

**Comment by kltm** [comment]
Note: https://github.com/geneontology/go-site/issues/313

**Comment by pgaudet** [comment]
This is cmungall here. I am assigning as showstopper for now. we can discuss f2f later if we want another mechanism for this.

**Comment by lpalbou** [comment]
@kltm could you check that the following ftp address will still be valid ? It is massively used in R/Bioconductor by the [GO.db package](http://bioconductor.org/packages/release/data/annotation/manuals/GO.db/man/GO.db.pdf):

> Mappings were based on data provided by: Gene Ontology ftp://ftp.geneontology.org/pub/go/godatabase/archive/latestlite/ With a date stamp from the source of: 2018-Oct10

**Comment by kltm** [comment]
@lpalbou Wow, that is an old old FTP address. As well, it's not been supported for some time, as far as I know. I think this may be a case where we go for the break and reach out to them if you think they were actually actively using it.

**Comment by lpalbou** [comment]
No idea but it's probably the good time to contact them. What seems strange however is that they were apparently able to retrieve a version on October 10 (copy paste from their R documentation) ?

**Comment by cmungall** [comment]
Process still running at Stanford, e.g
http://archive.geneontology.org/

Before contacting them we should have a plan

 1. Dockerize the mysql build code (a) and run as part of the pipeline (frequency tbd) (b)
 2. Volunteer help to migrate their code to existing downloads, potentially making new TSVs if it helps
 3. No support, just point them at existing files

1+2 would take resources, we'd coordinate this with go-managers as it would slow down other tasks. 1a could be done by anyone but tbh I could do it pretty quickly as it's all my code from aeons ago. 1b would exert some additional admin/processing cost. 2 would best be done by someone with R knowledge, Laurent or I can have Marcin help.

1 would be a reversal of our previous announcement but I'm sure 99% of the consumers of this never saw the announcement and would be happy to see it continue.

anyway short term fix is for them to point to archive.go but this too will go away.

**Comment by kltm** [comment]
@cmungall The archive walkback is still something in our TODO list (there's a ticket), but has a very low priority at this point.
As far as 1 goes, part of what we want to do with this transition is jettison as much of the legacy code and downloads as we can. I think that trying to bring in mysql support at this point is probably a bit much. (Although I could be argued into it.)
For 2, generating a new format that could be used in the mysql dump's stead would seem like a better way forward--there was a lot missing or unworkable in the old database and it might be good to take this chance to work with the community on a better export. As they are already pulling from a legacy system (e.g. no GO-CAMs, etc.). It may be easier to have such an export be materialized into an SQL format as well.
3 is always an option, but it might be good to get a little more information first.

**Comment by suzialeksander** [comment]
This is a list of minor issues, or just things that don't look right:

 * [ ] https://test.geneontology.io/docs/introduction-to-go-resource has link to https://geneontology.github.io/docs/go-enrichment-analysis/ , which seems to be at a different level of text zoom? I'd like to look at this again on a 'final' version of the site.

**Comment by suzialeksander** [comment]
Anyone have any problems with me consolidating https://test.geneontology.io/docs/go-file-format-guide/ onto https://test.geneontology.io/docs/downloads/, then deleting go-file-format-guide?

I think @pgaudet is still out, but @kltm ?

**Comment by lpalbou** [comment]
@suzialeksander what kind of problem are you having ?

The "Related Tools" section can go at the end of the /downloads file and the "GO Formats" can easily be merged in the paragraph "Download the Ontology" since this very short "Download the Ontology" is more a descriptive section of the files we provide rather than a way to download the files, except for the link "the Ontology" (in the title) which doesn't seem either clean or intuitive.

My proposal for the refactoring of those two pages:

## The GO Ontology
The ontology is released monthly and available in both [OBO](
http://owlcollab.github.io/oboformat/doc/obo-syntax.html) and [OWL](
https://github.com/owlcs/owlapi) formats in the [current GO release](http://current.geneontology.org/ontology). For convenience, an export in JSON is also available.

Files and programmatic ways to access those files are further described [here](https://test.geneontology.io/docs/download-ontology/) but we recommend two tools to parse, manipulate and serialize the ontology:
* [OBO format tools in GitHub](
https://github.com/oboformat/oboformat-tools): a wrapper for the Java (OWL-API) implementation of a parser for OBOF1.4 syntax and an implementation of the OBOF1.4 mapping to OWL (uses the OWL API)
* [OWL API in Github](https://github.com/owlcs/owlapi): a Java API for creating, manipulating and serialising OWL Ontologies.

## The GO Annotations
The annotations are released monthly, at the same time as the GO ontology and can also be retrieved from the [current GO release](http://current.geneontology.org/annotations). For convenience, we also provide taxon-specific release of the GO annotations in our [detailed download page](https://test.geneontology.io/docs/download-go-annotations).

GO annotations are provided in both [GAF](https://test.geneontology.io/docs/go-annotation-file-gaf-format-21) and [GPAD + GPI](https://test.geneontology.io/docs/gene-product-association-data-gpad-format/) formats.

**Comment by suzialeksander** [comment]
@lpalbou I meant problems as in "is anyone sentimentally attached to go-file-format-guide and would be angry if I deleted it" 😉 

Agreed, I don't really like the download link in the title, but the download link was previously the _entire title_ which was even worse.  Still straightening these pages out, thanks for the input!

**Comment by kltm** [comment]
@suzialeksander  No issues at my end--new stuff should be good on its own without history. Worst case is we have a page mapping for old stuff.

**Comment by cmungall** [comment]
+1 to consolidating

**Comment by kltm** [comment]
The super has gotten a little unwieldy here. All release-critical issues are to be spun out and put into the "In progress" column in the project. Clearing that column should be the prime immediate priority, and would effectively constitute the release once #958 cleared.

Talking with @suzialeksander I think for blockers, we have so far are:
#958 
#977
#978
#979 

Any other blockers should be added accordingly. Anything not a blocker should be bumped out of "In progress" for the moment.

**Comment by lpalbou** [comment]
@kltm you can remove #977 as it is done since October, although we could always improve it.

IMHO, all these posts should be in the geneontology.github.io repo and not here.

**Comment by lpalbou** [comment]
I am marking this as done as I didn't found any reference to that link: 
* Broken link: https://test.geneontology.io/docs/collaborations/cjmungall@lbl.gov on https://test.geneontology.io/docs/collaborations/

Also fixed rapidly some of the items

**Comment by kltm** [comment]
Yeah, it doesn't seem to be there any more. We'll likely spider periodically and can pick it up again if it shows up again.

**Comment by suzialeksander** [comment]
[ ] Still hard links to http://geneontology.org/external2go, eg http://geneontology.org/external2go/um-bbd_enzymeid2go : see https://test.geneontology.io/docs/download-mappings/: 

This item is waiting on #688, to see what the URLs look like after the mapping issues are resolved.

**Comment by suzialeksander** [comment]
Decided EuropePMC link is OK as it's not obvious how to make that search work in Pubmed.  Ok to close this ticket, I guess, after the above download-mappings links are double checked.
